src/Features/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -39,6 +45,69 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
             }
         }
 
+        /// <inheritdoc/>
+        protected override IMethodHandler GetHandlerForRequest(IQueueItem<RequestContext> work)
+        {
+            var defaultHandlerMetadata = new RequestHandlerMetadata(work.MethodName, work.RequestType, work.ResponseType, LanguageNames.CSharp);
```

Comment:
what ensures this method is only called serially (otherwise, this is not concurrency safe).